generator/internal/discovery_type_vertex_test.cc

```diff
@@ -763,6 +764,436 @@ message ReservedHighest {
+}
+
+TEST_F(DiscoveryTypeVertexDescriptorTest,
+       JsonToProtobufMessageWithDeletedFieldAndNewField) {
+  auto constexpr kPreviousProto = R"""(
+syntax = "proto3";
+package test.package;
+// Description of the message.
+message TestSchema {
+  reserved 3;
+  // Description for existingField.
+  optional string existing_field = 1;
+
+  // Description for deletedField.
+  repeated string deleted_field = 2;
+}
+)""";
+
+  auto constexpr kSchemaJson = R"""(
+{
+  "description": "Description of the message.",
+  "type": "object",
+  "id": "TestSchema",
+  "properties": {
+    "existingField": {
+      "type": "string",
+      "description": "Description for existingField."
+    },
+    "newField": {
+      "type": "array",
+      "description": "Description for newField.",
+      "items": {
+        "type": "string"
+      }
+    }
+  }
+}
+)""";
+
+  auto constexpr kExpectedProto = R"""(// Description of the message.
+message TestSchema {
+  reserved 2, 3;
+  // Description for existingField.
+  optional string existing_field = 1;
+
+  // Description for newField.
+  repeated string new_field = 4;
+}
+)""";
+
+  ASSERT_TRUE(AddProtoFile("test.proto", kPreviousProto));
+  auto json = nlohmann::json::parse(kSchemaJson, nullptr, false);
+  ASSERT_TRUE(json.is_object());
+  DiscoveryTypeVertex t("TestSchema", "test.package", json, &pool());
+  std::map<std::string, DiscoveryTypeVertex> types;
+  types.emplace("Foo", DiscoveryTypeVertex{"Foo", "test.package", {}, &pool()});
+  auto result = t.JsonToProtobufMessage(types, "test.package");
+  ASSERT_STATUS_OK(result);
+  EXPECT_THAT(*result, Eq(kExpectedProto));
+}
+
+TEST_F(DiscoveryTypeVertexDescriptorTest,
+       JsonToProtobufMessageWithBreakingTypeChange) {
+  auto constexpr kPreviousProto = R"""(
+syntax = "proto3";
+package test.package;
+// Description of the message.
+message TestSchema {
+  reserved 1;
+  // Description for existingField.
+  optional string existing_field = 2;
+}
+)""";
+
+  auto constexpr kSchemaJson = R"""(
+{
+  "description": "Description of the message.",
+  "type": "object",
+  "id": "TestSchema",
+  "properties": {
+    "newField": {
+      "$ref": "Foo",
+      "description": "Description for newField."
+    },
+    "existingField": {
+      "type": "number",
+      "format": "double",
+      "description": "Description for existingField."
+    }
+  }
+}
+)""";
+
+  ASSERT_TRUE(AddProtoFile("test.proto", kPreviousProto));
+  auto json = nlohmann::json::parse(kSchemaJson, nullptr, false);
+  ASSERT_TRUE(json.is_object());
+  DiscoveryTypeVertex t("TestSchema", "test.package", json, &pool());
+  std::map<std::string, DiscoveryTypeVertex> types;
+  types.emplace("Foo", DiscoveryTypeVertex{"Foo", "test.package", {}, &pool()});
+  auto result = t.JsonToProtobufMessage(types, "test.package");
+  EXPECT_THAT(result,
+              StatusIs(StatusCode::kInvalidArgument,
+                       HasSubstr("Message: test.package.TestSchema has field: "
```

Comment:
I assume this will happen sooner or later.  Are we supposed to manually patch the original proto and then regenerate?